rasterio/env.py

```diff
@@ -19,7 +20,8 @@
 # Rasterio defaults
 default_options = {
     'CHECK_WITH_INVERT_PROJ': True,
-    'GTIFF_IMPLICIT_JPEG_OVR': False
+    'GTIFF_IMPLICIT_JPEG_OVR': False,
+    "I'M_ON_RASTERIO": True
```

Comment:
Are there any implications with for GDAL? I know some invalid config values can cause crashes (https://github.com/mapbox/rasterio/issues/880) and I'm not entirely certain if arbitrary config options would be 100% safe.